packages/core/src/ceramic.ts

```diff
@@ -634,6 +634,13 @@ export class Ceramic implements CeramicApi {
       let stream
       try {
         if (query.genesis) {
+          if (
+            StreamUtils.isSignedCommitContainer(query.genesis) ||
+            StreamUtils.isSignedCommit(query.genesis)
+          ) {
+            return Promise.reject('Given genesis commit is not deterministic')
```

Comment:
```suggestion
            throw new Error('Given genesis commit is not deterministic')
```

Reply:
Better for the function's promise to resolve in an Error object rather than a string

Reply:
same as below